jenkins/automation/release-process.groovy

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env groovy
+pipeline {
+
+    agent {
+        node {
+            label 'docker-centos-7.9.2009-node'
+        }
+    }
+
+    parameters {
+        string(name: 'BASE_TAG', defaultValue: '', description: 'Image tag which need to be tagged with TARGET_TAG. It is used to tag component commits and create GitHub Release as well')
+        string(name: 'TARGET_TAG', defaultValue: '', description: 'New tag to be tagged on Image')
+        string(name: 'SERVICES_RELEASE', defaultValue: '', description: 'Services(cortx-k8s) release version on which image deployment is tested')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for tagging/Changelog/GitHub Release scripts', trim: true)
+        string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repository for tagging/Changelog/GitHub Release scripts', trim: true)
+    }
+
+    environment {
+        GITHUB_TOKEN = credentials('cortx-admin-token')
+        EMAIL_RECIPIENTS = "DEBUG"
+        REGISTRY = "ghcr.io"
+        OWNER = "seagate"
+        LATEST_GH_SERVER_IMAGE_TAG = sh( script: """
+            curl -s -H \"Accept: application/vnd.github+json\" -H \"Authorization: token ${GITHUB_TOKEN}\" \"https://api.github.com/repos/${OWNER}/cortx/releases/latest\" | jq '.tag_name' | tr -d '\"'
+        """, returnStdout: true).trim()
+        CORTX_SERVER_IMAGE = "${REGISTRY}/${OWNER}/cortx-rgw:${BASE_TAG}"
+        LATEST_GH_SERVER_IMAGE = "${REGISTRY}/${OWNER}/cortx-rgw:${LATEST_GH_SERVER_IMAGE_TAG}"
+        
+    }
+
+    stages {
+        stage("Tag Images") {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                script {
```

Comment:
Can we execute scripts instead of calling downstream jobs. Downstream jobs do not provide logs in on a single location.

Reply:
We cannot as tag image job does not have only one step to execute image tagging script. it has other stages as well to set env and if we try to execute things from here, we need to make changes in this job as well if parameters added or changed.